### PR TITLE
Adding queues stuff

### PR DESCRIPTION
**Pull request by jmatth**

### CODE REVIEW

**Comment by jmatth** [comment]
Hold off on this, it only supports GET requests on one endpoint so far.

File: queues/queue.go [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+	"errors"
+	"fmt"
+	"net/http"
+	"regexp"
+
+	"github.com/natebrennand/twiliogo/common"
+)
+
+const (
+	queuesURL = "https://api.twilio.com/2010-04-01/Accounts/%s/Queues/%s.json" // takes an AccountSid & QueueSid
```
Comment (natebrennand) [inline queues/queue.go]: rename to `getURL` to follow other packages

File: queues/queue.go [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+	CurrentSize     int             `json:"current_size"`
+	AverageWaitTime int             `json:"average_wait_time"`
+	MaxSize         int             `json:"max_size"`
+	DateCreated     common.JSONTime `json:"date_created"`
+	DateUpdated     common.JSONTime `json:"date_updated"`
+	Uri             string          `json:"uri"`
+}
+
+func validateQueueSid(sid string) bool {
+	match, _ := regexp.MatchString(`^(QU)[0-9a-z]{32}$`, string(sid))
```
Comment (natebrennand) [inline queues/queue.go]: unnecessary string cast

**Comment by natebrennand** [comment]
at some point if looks like we fucked up the `.travis.yml` and removed the step where it actually calls `go test ./...`.....
the build is actually broken on this...

**Comment by chelpu** [comment]
:frowning: I'll run tests and find the issues then.

**Comment by natebrennand** [comment]
Make sure to add the line to the .travis.yml too. Sorry for messing this up

**Comment by chelpu** [comment]
So I ran test ./... and there was no output... are you sure we don't have the line in .travis.yml?

**Comment by natebrennand** [comment]
weird, I get this when I run it. (are you definitely on master?)

```
~/g/s/g/n/twiliogo ❯❯❯ go test ./...
# github.com/natebrennand/twiliogo/examples/check_apps
examples/check_apps/main.go:30: undefined: applications.NewResource
ok      github.com/natebrennand/twiliogo    0.030s
ok      github.com/natebrennand/twiliogo/accounts   0.006s
ok      github.com/natebrennand/twiliogo/accounts/numbers   0.023s
ok      github.com/natebrennand/twiliogo/applications   0.006s
ok      github.com/natebrennand/twiliogo/common 0.027s
ok      github.com/natebrennand/twiliogo/conference 0.024s
# github.com/natebrennand/twiliogo/examples/make_call
examples/make_call/main.go:17: undefined: voice.Post
# github.com/natebrennand/twiliogo/examples/update_call
examples/update_call/main.go:16: undefined: voice.Post
```

I get this on the queue branch:

```
~/g/s/g/n/twiliogo ❯❯❯ go test ./...
# github.com/natebrennand/twiliogo/examples/check_apps
examples/check_apps/main.go:30: undefined: applications.NewResource
ok      github.com/natebrennand/twiliogo    0.031s
ok      github.com/natebrennand/twiliogo/accounts   0.009s
ok      github.com/natebrennand/twiliogo/accounts/numbers   0.029s
ok      github.com/natebrennand/twiliogo/applications   0.006s
ok      github.com/natebrennand/twiliogo/common 0.028s
ok      github.com/natebrennand/twiliogo/conference 0.025s
# github.com/natebrennand/twiliogo/examples/make_call
examples/make_call/main.go:17: undefined: voice.Post
# github.com/natebrennand/twiliogo/examples/update_call
examples/update_call/main.go:16: undefined: voice.Post
# github.com/natebrennand/twiliogo/queues
queues/queue_test.go:57: acct.getQueue undefined (type Account has no field or method getQueue)
```

**Comment by chelpu** [comment]
I didn't push my changes to queues_test.go so that would explain it. Will write some tests and push again.

**Comment by chelpu** [comment]
Should be good now.

**Comment by natebrennand** [comment]
fixed an example program that was causing `go test ./...` to fail for me locally because it couldn't compile

:+1:

**Comment by natebrennand** [comment]
that build report made no sense, restarting it

**Comment by natebrennand** [comment]
wtf, now it's green, but you can see that `golint` clearly broke midway and panicked

**Comment by chelpu** [comment]
Are there still issues with the build? Was golint's panic my fault or was that just a temporary problem with golint?

**Comment by natebrennand** [comment]
i think the master branch (default used for installation) of `golint` is broken. works fine for me locally.
feel free to merge

**Comment by natebrennand** [comment]
i say we ignore the error, it's `golint` that's broken, not our code. tests still pass

**Pull request merged**